src/transformers/models/vit/convert_vit_timm_to_pytorch.py

```diff
@@ -140,60 +137,70 @@ def convert_vit_checkpoint(vit_name, pytorch_dump_folder_path):
-            config.num_attention_heads = 3
-        elif vit_name[9:].startswith("small"):
-            config.hidden_size = 384
-            config.intermediate_size = 1536
-            config.num_hidden_layers = 12
-            config.num_attention_heads = 6
-        else:
-            pass
-    else:
-        if vit_name[4:].startswith("small"):
-            config.hidden_size = 768
-            config.intermediate_size = 2304
-            config.num_hidden_layers = 8
-            config.num_attention_heads = 8
-        elif vit_name[4:].startswith("base"):
-            pass
-        elif vit_name[4:].startswith("large"):
-            config.hidden_size = 1024
-            config.intermediate_size = 4096
-            config.num_hidden_layers = 24
-            config.num_attention_heads = 16
-        elif vit_name[4:].startswith("huge"):
-            config.hidden_size = 1280
-            config.intermediate_size = 5120
-            config.num_hidden_layers = 32
-            config.num_attention_heads = 16
 
     # load original model from timm
     timm_model = timm.create_model(vit_name, pretrained=True)
     timm_model.eval()
 
-    # load state_dict of original model, remove and rename some keys
+    # detect unsupported ViT models in transformers
+    # fc_norm is present
+    if not isinstance(getattr(timm_model, "fc_norm", None), torch.nn.Identity):
+        raise ValueError(f"{vit_name} is not supported in transformers because of the presence of fc_norm.")
+
+    # use of global average pooling in combination (or without) class token
+    if getattr(timm_model, "global_pool", None) == "avg":
+        raise ValueError(f"{vit_name} is not supported in transformers because of use of global average pooling.")
+
+    # CLIP style vit with norm_pre layer present
+    if "clip" in vit_name and not isinstance(getattr(timm_model, "norm_pre", None), torch.nn.Identity):
+        raise ValueError(
+            f"{vit_name} is not supported in transformers because it's a CLIP style ViT with norm_pre layer."
+        )
+
+    # SigLIP style vit with attn_pool layer present
+    if "siglip" in vit_name and getattr(timm_model, "global_pool", None) == "map":
+        raise ValueError(
+            f"{vit_name} is not supported in transformers because it's a SigLIP style ViT with attn_pool."
+        )
+
+    # use of layer scale in ViT model blocks
+    if not isinstance(getattr(timm_model.blocks[0], "ls1", None), torch.nn.Identity) or not isinstance(
+        getattr(timm_model.blocks[0], "ls2", None), torch.nn.Identity
+    ):
+        raise ValueError(f"{vit_name} is not supported in transformers because it uses a layer scale in its blocks.")
+
+    # Hybrid ResNet-ViTs
+    if not isinstance(timm_model.patch_embed, timm.layers.PatchEmbed):
+        raise ValueError(f"{vit_name} is not supported in transformers because it is a hybrid ResNet-ViT.")
+
+    # non-overlapping position and class token embedding (to be added)
```

Comment:
Yes, but I am not sure how to do it, maybe check if:
`torch.allclose(timm_model.cls_token[0, 0, :], timm_model.pos_embed[0, 0, :], atol=1e-3)`?

Reply:
right, that case, you can check an attribute on the timm model, `timm.model.no_embed_class` will be set to True when the cls embed doesn't overlap, although I think most models that use this would be caught by the layer scale check..

you can actually 'fix' this by prepending a zero value to the position embedding during conversion

Reply:
On all the `timm.list_pretrained('vit*')` models, the case when cls_embed does not overlap is not reached.
We can maybe add the check inside the layer scale case to have a fine-grained error message?

Reply:
@staghado it's used for deit3 (again w/ layer scale). It will also be used in a few new experiments coming out soon related to registers and global average pool pooling, so, would also be incompatible for other reasons... so can't think of any model that'd use no_embed_class without also using another unsupported feature...

Reply:
@rwightman Ok, I will remove it for now!